File: 프로그래머스/unrated/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.js
function solution(n) {
    if(n%2 === 0){
        return getSum(n, "pair");
    }else{
        return getSum(n, "solo");
    }
}

function getSum(number, type){
    let sum = 0;
    if(type === "solo"){
        for(let i = 1; i<=number; i++){
            if(i % 2 !== 0) sum += i;
        }
    }else{
        for(let i = 1; i<=number; i++){
            if(i % 2 === 0) sum +=i ** 2;
        }
    }
    
    return sum;
}

File: 프로그래머스/unrated/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.py
def solution(n):
    # 홀수인 경우 1부터 n까지 2씩 늘어남 따라서 홀수
    if n % 2 == 1:  
        return sum(i for i in range(1, n + 1, 2))
    # 짝수인 경우 2부터 n까지 2씩 늘어남 따라서 짝수 
    else:  
        return sum(i**2 for i in range(2, n + 1, 2))
